handler/src/main/java/io/netty/handler/ssl/SslUtils.java

```diff
@@ -478,7 +478,13 @@ static ByteBuf toBase64(ByteBufAllocator allocator, ByteBuf src) {
      * Validate that the given hostname can be used in SNI extension.
      */
     static boolean isValidHostNameForSNI(String hostname) {
+        /*
+         * These checks are based on sun.security.ssl.Utilities#rawToSNIHostName implementation
+         * [1] https://datatracker.ietf.org/doc/html/rfc6066#section-3
+         */
```

Comment:
We never did look at the OpenJDK implementation... So lets not say it here 
```suggestion
        // See  https://datatracker.ietf.org/doc/html/rfc6066#section-3
```